pkg/controllers/disruption/consolidation_test.go

```diff
@@ -608,69 +623,112 @@ var _ = Describe("Consolidation", func() {
-			nodeClaims := ExpectNodeClaims(ctx, env.Client)
-			nodes := ExpectNodes(ctx, env.Client)
-			Expect(nodeClaims).To(HaveLen(1))
-			Expect(nodes).To(HaveLen(1))
+				// should create a new nodeclaim as there is a cheaper one that can hold the pod
+				nodeClaims := ExpectNodeClaims(ctx, env.Client)
+				nodes := ExpectNodes(ctx, env.Client)
+				Expect(nodeClaims).To(HaveLen(1))
+				Expect(nodes).To(HaveLen(1))
 
-			// Expect that the new nodeclaim does not request the most expensive instance type
-			Expect(nodeClaims[0].Name).ToNot(Equal(nodeClaim.Name))
-			Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Has(v1.LabelInstanceTypeStable)).To(BeTrue())
-			Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Get(v1.LabelInstanceTypeStable).Has(mostExpensiveInstance.Name)).To(BeFalse())
+				// Expect that the new nodeclaim does not request the most expensive instance type
+				Expect(nodeClaims[0].Name).ToNot(Equal(nodeClaim.Name))
+				Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Has(v1.LabelInstanceTypeStable)).To(BeTrue())
+				Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Get(v1.LabelInstanceTypeStable).Has(mostExpensiveInstance.Name)).To(BeFalse())
 
-			// and delete the old one
-			ExpectNotFound(ctx, env.Client, nodeClaim, node)
-		})
-		It("can replace nodes if another nodePool returns no instance types", func() {
+				// and delete the old one
+				ExpectNotFound(ctx, env.Client, nodeClaim, node)
+			},
+			Entry("if the candidate is on-demand node", false),
+			Entry("if the candidate is spot node", true),
+		)
+		It("cannot replace spot node if less than minimum InstanceTypes flexibility", func() {
+			// Forcefully shrink the possible instanceTypes to be lower than 15 to replace a nodeclaim
+			cloudProvider.InstanceTypes = lo.Slice(fake.InstanceTypesAssorted(), 0, 5)
+			// Forcefully assign lowest possible instancePrice to make sure we have atleast one instance
+			// that is lower than the current node.
+			cloudProvider.InstanceTypes[0].Offerings[0].Price = 0.001
+			cloudProvider.InstanceTypes[0].Offerings[0].CapacityType = v1beta1.CapacityTypeSpot
+			spotInstances = lo.Filter(cloudProvider.InstanceTypes, func(i *cloudprovider.InstanceType, _ int) bool {
+				for _, o := range i.Offerings.Available() {
+					if o.CapacityType == v1beta1.CapacityTypeSpot {
+						return true
+					}
+				}
+				return false
+			})
+			// Sort the spot instances by pricing from low to high
+			sort.Slice(spotInstances, func(i, j int) bool {
+				return spotInstances[i].Offerings.Cheapest().Price < spotInstances[j].Offerings.Cheapest().Price
+			})
+			mostExpSpotInstance := spotInstances[len(spotInstances)-1]
+			mostExpSpotOffering := mostExpSpotInstance.Offerings[0]
 			labels := map[string]string{
 				"app": "test",
 			}
```

Comment:
nit: this might be the case with how this code ended up now, but you could cut down on the tests by defining this at the top level if you like.

Reply:
Addressed.